dashboards-observability/public/components/explorer/home_table/history_table.tsx

```diff
@@ -40,44 +40,100 @@ export function Table(options: TableData) {
   };
 
   const columns = [
+    {
+      field: 'type',
+      name: '',
+      width: '3%',
+      render: (item) => {
+        if (item == 'Visualization') {
+          return (
+            <div>
+              <EuiLoadingChart size="m" />
+            </div>
+          );
+        } else {
+          return (
+            <div>
+              <EuiIcon type="discoverApp" size="m" />
+            </div>
+          );
+        }
+      },
+    },
     {
       field: 'data',
       name: 'Name',
-      render: (item)=>{return <EuiLink onClick={() =>
-      {options.savedQuerySearch(item.query, [item.date_start, item.date_end], item.timestamp, item.fields)}}>
-        {item.name}
-      </EuiLink>},
+      align: 'left',
+      render: (item) => {
+        return (
+          <EuiLink
+            onClick={() => {
+              options.savedQuerySearch(
+                item.query,
+                [item.date_start, item.date_end],
+                item.timestamp,
+                item.fields
+              );
+            }}
+          >
+            {item.name}
+          </EuiLink>
+        );
+      },
     },
     {
-      field: 'description',
-      name: 'Description',
+      field: 'type',
+      name: 'Type',
+      align: 'right',
     },
   ];
 
-
-
+  let queryType = '';
   const queries = options.savedHistory.map((h) => {
     const savedObject = h.hasOwnProperty('savedVisualization')
       ? h.savedVisualization
       : h.savedQuery;
+    const isSavedVisualization = h.hasOwnProperty('savedVisualization');
+    if (isSavedVisualization) {
+      queryType = 'Visualization';
+    } else {
+      queryType = 'Query';
+    }
     return {
       data: {
         name: savedObject.name,
         query: savedObject.query,
         date_start: savedObject.selected_date_range.start,
-        date_end : savedObject.selected_date_range.end,
+        date_end: savedObject.selected_date_range.end,
         timestamp: savedObject.selected_timestamp?.name || '',
-        fields: savedObject.selected_fields?.tokens || []
+        fields: savedObject.selected_fields?.tokens || [],
       },
       name: savedObject.name || '',
-      description: savedObject.description || '',
-
+      type: queryType,
     };
   });
 
- 
   const totalItemCount = queries.length;
 
+  const search = {
+    box: {
+      incremental: true,
+    },
+    filters: [
+      {
+        type: 'field_value_selection',
+        field: 'type',
+        name: 'Type',
+        multiSelect: false,
+        options: FILTER_OPTIONS.map((i) => ({
+          value: i,
+          name: i,
+          view: `${i}`,
```

Comment:
Why is the `${i}` variable used like this? Can't it just be like the above lines? `i`

Reply:
Yeah it can be like above lines too